src/ui/components/radioSelector/RadioSelector.tsx

```diff
@@ -0,0 +1,128 @@
+import React, { useRef } from 'react'
+import styled, { DefaultTheme } from 'styled-components/native'
+
+import { useHandleFocus } from 'libs/hooks/useHandleFocus'
+import { accessibleRadioProps } from 'shared/accessibilityProps/accessibleRadioProps'
+import { TouchableOpacity } from 'ui/components/TouchableOpacity'
+import { useArrowNavigationForRadioButton } from 'ui/hooks/useArrowNavigationForRadioButton'
+import { useSpaceBarAction } from 'ui/hooks/useSpaceBarAction.web'
+import { AccessibleIcon } from 'ui/svg/icons/types'
+import { ValidateOff } from 'ui/svg/icons/ValidateOff'
+import { RadioButtonSelected } from 'ui/svg/RadioButtonSelected'
+import { getSpacing, Typo } from 'ui/theme'
+
+export enum RadioSelectorType {
+  DEFAULT = 'default',
+  ACTIVE = 'active',
+  DISABLED = 'disabled',
+}
+
+interface ValidateOffIconProps extends AccessibleIcon {
+  type?: RadioSelectorType
+}
+
+interface RadioSelectorProps {
+  label: string
+  onPress: (label: string, type: RadioSelectorType) => void
+  description?: string
+  price?: string
+  type?: RadioSelectorType
+}
+
+export const RadioSelector = ({
+  label,
+  onPress,
+  description,
+  price,
+  type = RadioSelectorType.DEFAULT,
+}: RadioSelectorProps) => {
+  const containerRef = useRef(null)
+  const { onFocus, onBlur, isFocus } = useHandleFocus()
+
+  const handlePress = () => {
```

Comment:
Il y a potentiellement une autre manière de gérer ça 

let onHandlePress = () => {onPress(label, RadioSelectorType.ACTIVE)}
if(type === RadioSelectorType.DISABLED) onHandlePress = null

Et plus bas tu appelles onHandlePress

Reply:
Cela dépend des préférences je suis plus première méthode.

La première méthode est plus concise et facile à lire, mais nécessite une condition supplémentaire pour gérer le cas où le type est DISABLED. On gère l'option `DISABLED` de manière explicite et cela permet de retourner une sortie cohérente en cas de désactivation.

La seconde méthode est plus complexe car elle utilise une affectation conditionnelle, ce qui peut rendre le code moins lisible et plus difficile à comprendre. Elle est tout de même  plus explicite sur ce qui se passe lorsque le type est DISABLED, mais peut nécessiter plus de lignes de code.

Reply:
Personnellement je n'aime pas la méthode que propose @pierrecanthelou.

Déjà on a un `let` et j'ai pour habitude de ne pas les aimer.
Ensuite je pars du principe où le handler doit toujours être défini, quitte à ce qu'il ne fasse rien. De cette façon ça évite de gérer le cas du `<Comp handler={null} />` qui est quand même bizarre. Autant `undefined` passe, autant `null` j'aime pas du tout :P 

Enfin, je ne vois pas l'intérêt que ça apporte. Au final j'ai l'impression que ça ne va faire que plus nous rendre confus.

Reply:
De mon côté je préfère aussi comme tu as fait Yassin.

Reply:
ATTENTION :
1) c'était un commentaire, pas une demande d'amélioration ;)
2) c'était plus pour montrer une autre manière de gérer la condition sans qu'elle soit à chaque fois dans le handle. (Alors ok pour un handle on s'en fiche, à moins que l'utilisateur s'excite sur le bouton, mais ça peut être important...)
3) pour coller à la première solution on ne mettrai pas "null" mais "() => {return}", et c'est pas joli
4) c'est un pattern assez commun qu'il me semblait intéressant de remonter.
5) pas de 5 mais vous auriez pu proposer un opérateur ternaire ;)

const onHandlePress = 
type === RadioSelectorType.DISABLED ? () => undefined : () => {onPress(label, RadioSelectorType.ACTIVE)}

Reply:
OK pour le ternaire à la limite, mais c'est évité je crois dans le code parce qu'ils n'aiment pas les single line :D 

En somme donc, pour coller avec les bonnes pratiques ainsi que celle du Pass (je fais la distinction :P) je pense que sa solution est celle qui est idéale :) 

Mais merci pour ton commentaire, ça peut être intéressant ! Effectivement il ne fait pas oublier que parfois, éviter de créer une fonction c'est bien.

Dans ce cas si on le voulait vraiment je serai parti sur 
```suggestion
const handlePress = type === RadioSelectorType.DISABLED
  ? undefined
  : () => {onPress(label, RadioSelectorType.ACTIVE)}
```
C'est à peu près ce que tu as fait au-dessus sauf que pour le cas de `DISABLED` je ne crée pas de fonction.

Et en vrai, écrit comme ça, je suis pas contre l'utilisation de cette méthode.